Pull Request

patrykgajda opened the pull request.

patrykgajda committed:
ClientModel - add done
PizzaModel - menu

patrykgajda committed:
MVC for Mirek

patrykgajda committed:
Done

MiroRat commented on src/main/java/VirtualTrattoria.java:
```diff
@@ -1,28 +1,44 @@
+        ClientService clientService = new ClientServiceImpl();
+
+        ClientController clientController = new ClientController();
+        clientController.setClientService(clientService); // couldn't figure out how can I fix list reset
```
I would add clientService to ClientControler constructor instead of using a setter as controller cannot work correctly without a service in that case.

MiroRat commented on src/main/java/VirtualTrattoria.java:
```diff
@@ -1,28 +1,44 @@
-        ClientOrder clientOrder = new ClientOrder();
-        Menu menu = new Menu();
-        menu.showMenu();
+        clientController.addClient();
```
I would add General/Main/App Controller with start() or startApp() method which would then trigger methods from client controller.

MiroRat commented on src/main/java/Controller/ClientController.java:
```diff
@@ -0,0 +1,30 @@
+    private ClientService clientService = new ClientServiceImpl();
+    private final AddClientView addClientView = new AddClientView();
+    private final GetClientView getClientView = new GetClientView();
+
```
Addition of new public method like 'processClient' can be considered. It would give us opportunity to make addClient and getAllClients both private and call them from processClient - just note to thing about it - does not need to be implemented.

MiroRat commented on src/main/java/Controller/ClientOrderController.java:
```diff
@@ -0,0 +1,78 @@
+    public void addClientOrder(){
+        clientOrderService.setClientService(clientService); // couldn't figure out how can I fix list reset
+        getClientView.printWhatIsYourNumber();
+        clientNo = scanner.nextInt();
```
possible solution to add exception which we discussed:

```
  boolean incorrectInput = true;
        while(incorrectInput) {
            try {
                incorrectInput = false;
                clientNo = scanner.nextInt();
            } catch (InputMismatchException e) {
                System.out.println("Incorrect value. Please enter a number");
                scanner.next();
                incorrectInput = true;
            }
        }
```

Explanation:
https://stackoverflow.com/questions/3572160/how-to-handle-infinite-loop-caused-by-invalid-input-inputmismatchexception-usi

MiroRat commented on src/main/java/Controller/ClientOrderController.java:
```diff
@@ -0,0 +1,78 @@
+        getClientView.printWhatIsYourNumber();
+        clientNo = scanner.nextInt();
+        // how does try - catch work I tried to use it here but it saves data and it block 2nd use of AddClientOrder
+        if (clientOrderService.clientNoValidation(clientNo)) {
```
As a exercise - this can also be changed to try catch

patrykgajda committed:
added few Tests with Junit

Ingredients update

Try Catch added

Added MainController

Services moved to Constructors

Ingredients > stream
